src/backend/gitalytics_api/cookies.py

```diff
@@ -84,8 +84,29 @@ async def endpoint(token: dbm.Session = session_from_cookies):
 
 
 @fastapi.Depends
-def active_workspace_id(cookie_storage: EncryptedCookieStorage = EncryptedCookieStorage) -> str:
+def active_workspace_id(cookie_storage: EncryptedCookieStorage = EncryptedCookieStorage) -> int:
     try:
         return cookie_storage[CookieKey.ACTIVE_WORKSPACE_ID]
     except KeyError:
-        raise fastapi.HTTPException(fastapi.status.HTTP_424_FAILED_DEPENDENCY, detail="no active workspace selected")
+        # raise fastapi.HTTPException(fastapi.status.HTTP_424_FAILED_DEPENDENCY, detail="no active workspace selected")
+        pass
+
+    try:
+        session_id = cookie_storage[CookieKey.SESSION_ID]
+    except KeyError:
+        raise fastapi.HTTPException(fastapi.status.HTTP_401_UNAUTHORIZED)
+
+    with createLocalSession() as connection:
+        workspace_id: int | None = connection.query(dbm.Workspace.id) \
+            .select_from(dbm.Session) \
+            .join(dbm.Repository, dbm.Session.repositories) \
+            .join(dbm.Workspace) \
+            .filter(dbm.Session.id == session_id) \
+            .limit(1) \
+            .scalar()
+
+    if workspace_id is None:
+        raise fastapi.HTTPException(fastapi.status.HTTP_425_TOO_EARLY)
```

Comment:
Ich verstehe wieso du das gemacht hast, und es ist keine schlechte idee, aber es hat einen großen Hacken.
Die Session-access-rights sind auf repositories gescoped. Wir dürfen also nicht davon ausgehen, dass wenn er einen workspace_id cookie hat (auch wenn er eine session_id hat, die zugriff auf ein repo in dem workspace hat), dass er dann zugriff auf ALLE repos in dem Workspace hat. 

Deswegen MUSS (leider) immer manuell auch die session übergeben werden, von der session die repos geholt werden, über die repos das workspace gefiltered werden. Denn die Session hat nicht unbedingt zugriff auf jedes Repo in dem Workspace.